### PR TITLE
tp(cdc): support manual move table and rebalance

### PR DESCRIPTION
**Pull request by 3AceShowHand**
### What problem does this PR solve?

Issue Number: ref #4757 

### What is changed and how it works?
2 phase scheduling scheduler support manual move table and rebalance.

#### Tests 

 - No code

### Release note 

```release-note
`None`
```

### CODE REVIEW

File: cdc/scheduler/internal/tp/scheduler_move_table.go [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+	if _, ok := m.tasks[tableID]; ok {
+		return false
+	}
+	m.tasks[tableID] = &scheduleTask{
+		moveTable: &moveTable{
+			TableID:     tableID,
+			DestCapture: target,
+		},
+		accept: func() {
+			delete(m.tasks, tableID)
```
Comment (overvenus) [inline cdc/scheduler/internal/tp/scheduler_move_table.go:49]: Could you make it thread-safe? Even though it's single threaded now, I'd like to make it decoupled with other components implementation.

File: cdc/processor/processor_test.go [hunk 2]
```diff
@@ -268,6 +268,9 @@ func TestTableExecutorAddingTableDirectly(t *testing.T) {
 	table1 := p.tables[1].(*mockTablePipeline)
 	require.Equal(t, model.Ts(20), table1.sinkStartTs)
 	require.Equal(t, pipeline.TableStateReplicating, table1.status)
+	meta := p.GetTableMeta(model.TableID(1))
```
Comment (overvenus) [inline cdc/processor/processor_test.go:271]: Please remove `TestGetTableMeta` in processor_test.go.

File: cdc/scheduler/internal/tp/agent.go [hunk 3]
```diff
@@ -116,7 +116,7 @@ func NewAgent(ctx context.Context,
 				zap.String("namespace", changeFeedID.Namespace),
 				zap.String("changefeed", changeFeedID.ID),
 				zap.Error(err))
-			return nil, nil
+			return result, nil
```
Comment (overvenus) [inline cdc/scheduler/internal/tp/agent.go:119]: Is there any test cover the change?

**Comment by overvenus** [comment]
Please fix make check, see verify ci.

File: cdc/scheduler/internal/tp/agent.go (hunk 3 above)
Comment (3AceShowHand) [reply cdc/scheduler/internal/tp/agent.go:119]: no, `NewAgent` is based on `etcdClient` and other stateful parameters, does not have ut yet.

**Review by overvenus — APPROVED** [review]

**Pull request merged**